### ISSUE TITLE
`jinja2.Markup` no longer exists as of `3.1.0`

### ISSUE THREAD

**Issue by mhostetter** [jbms/sphinx-immaterial#49]
As of this morning, I get a failing build on RTD with this error. Upon investigation, the latest `jinja2` (released today) does not support `jinja2.Markup`, see [here](https://jinja.palletsprojects.com/en/3.1.x/changes/?highlight=markup#version-3-1-0). Instead, it states `Markup` should be imported from `markupsafe`.

I can submit a PR to either:
* Pin `jinja < 3.1.0`, or
* Add the `markupsafe` dependency (I believe it's already required by `jinja2`) and change the code to import `Markup` accordingly

Thoughts?

```
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/events.py", line 101, in emit
    results.append(listener.handler(self.app, *args))
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx_immaterial/nav_adapt.py", line 410, in _html_page_context
    page_title = jinja2.Markup.escape(jinja2.Markup(context.get("title")).striptags())
AttributeError: module 'jinja2' has no attribute 'Markup'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/cmd/build.py", line 284, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/application.py", line 337, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 294, in build_update
    self.build(to_build,
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 358, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 532, in write
    self._write_serial(sorted(docnames))
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 542, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/builders/html/__init__.py", line 632, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/builders/html/__init__.py", line 1033, in handle_page
    newtmpl = self.app.emit_firstresult('html-page-context', pagename,
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/application.py", line 464, in emit_firstresult
    return self.events.emit_firstresult(event, *args,
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/events.py", line 119, in emit_firstresult
    for result in self.emit(name, *args, allowed_exceptions=allowed_exceptions):
  File "/home/docs/checkouts/readthedocs.org/user_builds/galois/envs/312/lib/python3.8/site-packages/sphinx/events.py", line 109, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function _html_page_context at 0x7f03eacd1a60> for event 'html-page-context' threw an exception (exception: module 'jinja2' has no attribute 'Markup')

Extension error (sphinx_immaterial.nav_adapt):
Handler <function _html_page_context at 0x7f03eacd1a60> for event 'html-page-context' threw an exception (exception: module 'jinja2' has no attribute 'Markup')
```

**Comment by jbms** [comment]
Thanks for raising this issue.  I think it would be best to add the dependency on markupsafe and avoid pinning jinja2.

**Comment by 2bndy5** [comment]
We should push a tag soon (for actual pypi index). I just got an error from an RTD build (unrelated to this repo) about this issue.

**Comment by 2bndy5** [comment]
fix included in v0.4.1
